cargo-cyclonedx/src/generator.rs

```diff
@@ -169,6 +171,42 @@ impl SbomGenerator {
         component
     }
 
+    fn get_purl(package: &Package) -> Result<Purl, purl::PackageError> {
```

Comment:
I think this method needs unit tests

Reply:
I'm reasonably confident that it is correct given my assumptions of how `cargo metadata` works. What it needs is integration tests to test it against real `cargo metadata` rather than synthetic data I made up, which is still limited by my assumptions.